curvefs/src/tools/create/curvefs_create_topology_tool.cpp

```diff
@@ -628,7 +628,9 @@ int CurvefsBuildTopologyTool::CreateServer() {
     for (auto it : serverDatas) {
         ServerRegistRequest request;
         ServerRegistResponse response;
-        request.set_hostname(it.name);
+        if (!it.name.empty()) {
+            request.set_hostname(it.name);
```

Comment:
You need change hostName of ServerRegistRequest in topology.topo to optional.

Reply:
ok